include/micm/solver/rosenbrock.hpp

```diff
@@ -70,10 +70,10 @@ namespace micm
     size_t number_of_grid_cells_{ 1 };  // Number of grid cells to solve simultaneously
   };
 
-  /**
-   * @brief An implementation of the Chapman mechnanism solver
-   *
-   */
+   /// @brief An implementation of the Chapman mechnanism solver
+   ///
+   /// The template parameter is the type of matrix to use
+  template<template<class> class M>
```

Comment:
Would it be possible to replace `M` with `Matrix` everywhere, or is that not possible because of our own `Matrix` class?

Reply:
Yeah, I think it might at least be confusing to use `Matrix`, even if it is technically possible. In almost all the code I've seen that uses template parameters, the parameters are a single capital letter. Normally, I'll look at the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html) for something like this, but I can't find anywhere where they specifically talk about template parameter naming.

I'm in favor of adopting a style for template parameter naming, but don't have a strong preference for what we choose.  Maybe we can choose a couple options and do pros/cons?

### Single capital letter
**Pros**: Easy to identify template parameters in code (if this is the only thing we allow to be a single capital letter). Consistent with many examples of template code.

**Cons**: Is not descriptive and might become confusing if multiple template parameters start to be used for a single class or function.

Reply:
How would you feel about `MatrixPolicy`? This indicates that the use of a thing is meant to provide different behavior, not just generation of code for a different data type. Behavior changes will have a suffix of `Policy` and data types will be a single letter?

Reply:
I like that solution. Will make the change now

Reply:
Ok, another new terminology to me. What is `Policy` here?

Reply:
It's a strategy for treating things by behaviors (like what functions they implement): https://en.wikipedia.org/wiki/Modern_C%2B%2B_Design

For our `AddForcingTerms` function, the `MatrixPolicy` type just has to have the `[]` syntax for 2D arrays (it should probably work if you passed it a `vector<vector<double>>`). Our specialized `AddForcingTerms` function that works with `VectorMatrix` gets used for things that have the additional functions that `VectorMatrix` provides:
```
  /// Concept for vectorizable matrices
  template<typename T>
  concept Vectorizable = requires(T t) {
    t.BlockSize();
    t.NumberOfBlocks();
    t.VectorSize();
  };

...

  template<template<class> typename MatrixPolicy>
    requires Vectorizable<MatrixPolicy<double>>
  void AddForcingTerms(const MatrixPolicy<double>& rate_constants, const MatrixPolicy<double>& state_variables, MatrixPolicy<double>& forcing)
      const;
```
So, the specialized function works with `VectorMatrix` and any other type that has the `BlockSize()`, `NumberOfBlocks()` and `VectorSize()` functions